### PR TITLE
Initial work newsletter signup form

### PR DESCRIPTION
**Pull request by DipakShrestha18**

### CODE REVIEW

File: src/components/NewsletterForm/NewsletterForm.tsx [hunk 1]
```diff
@@ -14,13 +15,18 @@ import NewsletterFormError from './NewsletterFormError';
...
   className?: string;
+  type?: string;
+  researchOption?: string;
 };
 
 export const NewsletterForm = ({
   children,
-  className
+  className,
+  type,
```
Comment (creido-welly) [inline src/components/NewsletterForm/NewsletterForm.tsx]: Minor: put these in alphabetical order

File: src/components/NewsletterForm/NewsletterForm.tsx [hunk 2]
```diff
@@ -75,7 +81,9 @@ export const NewsletterForm = ({
     const response = await fetchNewsletterResponse(
       // TODO: #6023 - move to .env
       'https://wellcome.ac.uk/newsletter-signup',
-      email
+      email,
```
Comment (creido-welly) [inline src/components/NewsletterForm/NewsletterForm.tsx:85]: Minor: alphabetical order

File: src/components/NewsletterForm/NewsletterFormItem.tsx [hunk 3]
```diff
@@ -4,7 +4,7 @@ import cx from 'classnames';
 type NewsletterFormItemProps = {
   children: React.ReactNode;
   hasError?: boolean;
-  type?: 'email' | 'consent' | 'submit' | 'footer' | 'error';
+  type?: 'email' | 'dropdown' | 'consent' | 'submit' | 'footer' | 'error';
```
Comment (creido-welly) [inline src/components/NewsletterForm/NewsletterFormItem.tsx]: Minor: alphabetical order

File: src/components/NewsletterForm/NewsletterFormResearchDropDown.tsx [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+export const NewsletterFormResearchDropDown = ({
+  handleChange,
+  value
+}: NewsletterFormResearchDropDownProps) => {
+  return (
+    <NewsletterFormItem type="dropdown">
+      <label
+        className="newsletter-form__item-label newsletter-form__item-label--dropdown"
+        htmlFor="dropdown"
```
Comment (creido-welly) [inline src/components/NewsletterForm/NewsletterFormResearchDropDown.tsx]: `htmlFor` is react equivalent of `for` attribute in HTML and so should reference the id of the associated component

in this case this should be `edit-research-dd`

File: src/components/NewsletterForm/fetch-newsletter-response.ts [hunk 5]
```diff
@@ -1,11 +1,15 @@
 export const fetchNewsletterResponse = async (
   endpointUrl: string,
-  email: string
+  email: string,
```
Comment (creido-welly) [inline src/components/NewsletterForm/fetch-newsletter-response.ts:2]: Minor: alphabetical

File: src/components/NewsletterForm/NewsletterFormResearchDropDown.tsx [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+      >
+        Which of these best describes you?
+      </label>
+      <select
+        id="edit-research-dd"
```
Comment (creido-welly) [inline src/components/NewsletterForm/NewsletterFormResearchDropDown.tsx]: `-options` makes more sense than `-dd` as a suffix. dd is also coupling the type to a particular form element which could potentially change

**Comment by DipakShrestha18** [comment]
@creido-welly changes made, re-review pls.

File: src/components/NewsletterForm/NewsletterFormResearchDropDown.tsx [hunk 7]
```diff
@@ -15,12 +15,12 @@ export const NewsletterFormResearchDropDown = ({
     <NewsletterFormItem type="dropdown">
       <label
         className="newsletter-form__item-label newsletter-form__item-label--dropdown"
-        htmlFor="dropdown"
+        htmlFor="edit-research-dd"
```
Comment (creido-welly) [inline src/components/NewsletterForm/NewsletterFormResearchDropDown.tsx]: this needs to be updated to edit-research-options too

File: src/components/NewsletterForm/NewsletterFormResearchDropDown.tsx [hunk 8]
```diff
@@ -15,12 +15,12 @@ export const NewsletterFormResearchDropDown = ({
...
       >
         Which of these best describes you?
       </label>
       <select
-        id="edit-research-dd"
+        id="edit-research-options"
         name="research_dd"
```
Comment (creido-welly) [inline src/components/NewsletterForm/NewsletterFormResearchDropDown.tsx]: if we update the id we may as well do the same with the name i.e. research options makes more sense than research dd

**Comment by DipakShrestha18** [comment]
@creido-welly made changes as suggested.

File: src/components/NewsletterForm/fetch-newsletter-response.ts [hunk 9]
```diff
@@ -1,11 +1,15 @@
...
-  email: string
+  researchOption?: string,
+  type?: string
 ) => {
   const requestOptions = {
     method: 'POST',
     body: JSON.stringify({
-      email
+      email,
+      research_dd:researchOption,
```
Comment (creido-welly) [inline src/components/NewsletterForm/fetch-newsletter-response.ts:11]: should this now be research_options?

**Review by creido-welly — APPROVED** [review]
One query to answer otherwise good to go

File: src/components/NewsletterForm/fetch-newsletter-response.ts (hunk 9 above)
Comment (DipakShrestha18) [reply src/components/NewsletterForm/fetch-newsletter-response.ts:11]: I think for consistency researchOption is good as we are doing camel case in props.

**Pull request merged**